pyroma/projectdata.py

```diff
@@ -200,7 +200,7 @@ def run_setup(script_name, script_args=None, stop_after="run"):
 
     if core._setup_distribution is None:
         raise RuntimeError(
-            "'distutils.core.setup()' was never called -- " f"perhaps '{script_name}' is not a Distutils setup script?"
+            f"'distutils.core.setup()' was never called -- perhaps '{script_name}' is not a Distutils setup script?"
```

Comment:
I missed this before, but as someone brought up on another CPython PR, `'{spam}'` is considered an antipattern and its better to do:

```suggestion
            f"'distutils.core.setup()' was never called -- perhaps {script_name!r} is not a Distutils setup script?"
```

Reply:
Please could you explain more? Do you have a link to the other PR?

Reply:
Sorry I didn't link it initially, I had to go digging, but here you go: https://github.com/python/cpython/pull/95749#discussion_r947897759

Ezio didn't provide a lot of detail, but I've always thought and done the same since it is simpler, more consistent and less error prone than manually inserting the quotes, ensuring they go in the right places and are the right quote type for the string in question. We could ping him if you're curious for his rationale (which is probably clearer/stronger than mine).

Reply:
I did some testing around, and yeah, using `{var!r}` has the benefit that you will not end up with things like:
```
>>> strung = "Don't"
>>> print(f"Do not write '{strung}' inside quotes")
Do not write 'Don't' inside quotes
```
But that's really not relevant here, as `script_name` won't have accents. But it's a neat thing I haven't thought about before.